### DISCUSSION TITLE
Make library compilable on user's end

### DISCUSSION THREAD

**Discussion by g-cheishvili** [SAP/ui5-webcomponents-ngx#69] [Ideas]
Currently what we do is that we parse the ui5 on our end, then create wrappers and build them, then publish them on npm, this is a quick way for users to use our wrappers, but it also has few drawbacks:
- User is forced to use whatever `@angular/cli` version we have
- User has to install a specific version of the `@ui5/*`, the same versions as we do

What I think we could do is that we can ship just the parser and a generator and have `postinstall` hook, which would generate the library on user's application, effectively making it support Angular versions all the way from `14`! Also, users would be able to install whatever the ui5 version they have and also we could add support for prefixed components, without a lot of effort.

**Comment by petermuessig** [comment]
The approach sounds interesting. Maybe this could be even done in two flavors, a pre-built variant and the one to be integrated in your project (for experts). The pre-built variant has the charm to definitely work but in case of having floating versions for Angular and UI5 WebC this may lead to build issues.

**Reply by g-cheishvili** [reply]
For allowing that we would need to release the tools which we use for our builds though, but if released then yes, we could do that no problem at all. We could even allow other `@ui5` based libraries to be built inside any other project

**Reply by g-cheishvili** [reply]
I just did not like how ng16 update had an effect on us. We had to make breaking changes in order to update, but technically for our library there is nothing that stops us to support anything from ng-14
